src/VehicleSetup/JoystickConfigButtons.qml

```diff
@@ -52,63 +52,146 @@ ColumnLayout {
+
+                        function _setButtonPwm(button, isLow, pwm) {
                             if(_activeJoystick) {
-                                var i = find(_activeJoystick.buttonActions[modelData])
-                                if(i < 0) i = 0
-                                currentIndex = i
+                                if (pwm < 1000) {
+                                    pwm = 1000;
+                                }
+                                if (pwm > 2000) {
+                                    pwm = 2000;
+                                }
+                                _activeJoystick.setButtonPwm(modelData, isLow, pwm)
                             }
                         }
 
-                        Component.onCompleted:  _findCurrentButtonAction()
-                        onModelChanged:         _findCurrentButtonAction()
-                        onActivated:            _activeJoystick.setButtonAction(modelData, textAt(index))
-                    }
-                    QGCCheckBox {
-                        id:                         repeatCheck
-                        text:                       qsTr("Repeat")
-                        enabled:                    currentAssignableAction && _activeJoystick.calibrated && currentAssignableAction.canRepeat
-                        onClicked: {
-                            _activeJoystick.setButtonRepeat(modelData, checked)
-                        }
-                        Component.onCompleted: {
+                        function _getButtonPwm(button, isLow) {
+                            var pwm = -1;
                             if(_activeJoystick) {
-                                checked = _activeJoystick.getButtonRepeat(modelData)
+                                pwm = _activeJoystick.getButtonPwm(modelData, isLow)
+                            }
+                            return pwm == -1 ? "" : pwm;
+                        }
+
+                        QGCLabel {
+                            id:         lowPwmLabel
+                            text:       qsTr("Low")
+                            anchors.verticalCenter:     parent.verticalCenter
+                        }
+                        QGCTextField {
+                            width:  ScreenTools.defaultFontPixelWidth * 10
+                            implicitHeight:     ScreenTools.implicitTextFieldHeight
+                            visible:                true
+                            validator: IntValidator { bottom:1000; top: 2000}
+
+                            Component.onCompleted: {
+                                if(_activeJoystick) {
+                                    text = parent._getButtonPwm(modelData, true)
+                                }
+                            }
+                            onEditingFinished: parent._setButtonPwm(modelData, true, text)
```

Comment:
This text field needs to update when the channel selection changes. For example if you set it up for Channel 10, and then use the drop down to select Channel 11, the PWM values in "low" and "high" will not get applied but they still show up in the text fields. These text fields should reset to 0 when the channel selection changes.